src/MIDebugEngine/Natvis.Impl/Natvis.cs

```diff
@@ -1040,12 +1040,19 @@ private string ReplaceNamesInExpression(string expression, IVariableInformation
         {
             return ProcessNamesInString(expression, new Substitute[] {
                 (m)=>
-                    {   // finds children of this structure and sub's in the fullname of the child
-                        IVariableInformation var = variable == null ? null : variable.FindChildByName(m.Value);
-                        if (var != null)  // found a child
-                        {
-                            return "(" + var.FullName() + ")";
-                        }
+                    {
+                        if (variable == null)
+                            return null;
+
+                        // replace explicit this references
+                        if (m.Value == "this")
+                            return "(&" + variable.FullName() + ")";
```

Comment:
Not sure if additional parentheses or special handling for corner cases like pointers might be necessary here.